DFE-Digital/early-careers-framework
# Add API endpoint for DQT data

### Context

- NPQ Registration requires a hard TRN validation check which can be resolved by fetching data from DQT

### Changes proposed in this pull request

- Builds upon @ethanmills spike work
- Adds api endpoint which allows clients to fetch data from DQT
- Minor tweak to fix rspec deprecation warning

### Guidance to review

- `ParticipantValidationService` was introduced in an earlier commit but may not fit our needs as we do not collect NI number at the moment but may change in the future
- As a result the API endpoint just proxies the data from DQT so clients can meet whatever needs they may have

### Testing

- Setup DQT api keys so it can call into the sandbox
- Create an api token with `NpqRegistrationApiToken.create_with_random_token!`
- Note down the unhashed token, it will be needed later
- Call endpoint with `curl -H "Authorization: Bearer UNHASHED_TOKEN" localhost:3001/api/v1/dqt-records/VALID_TRN`
- See spreadsheet of valid TRNs present in dataset
- Should return a json response with some data

### Review Checks

### How can I view this in a review app?

- NOTE: not sure if review apps are hooked up to DQT sandbox? if not this won't work
- Create an api token with `NpqRegistrationApiToken.create_with_random_token!`
- Note down the unhashed token, it will be needed later
- Call endpoint with `curl -H "Authorization: Bearer UNHASHED_TOKEN" localhost:3001/api/v1/dqt-records/VALID_TRN`
- See spreadsheet of valid TRNs present in dataset
- Should return a json response with some data
